tests/framework/filters/AccessRuleTest.php

```diff
@@ -88,7 +129,55 @@ public function testMatchAction()
 
     // TODO test match controller
 
-    // TODO test match roles
+    public function testMatchRole()
+    {
```

Comment:
because of the repetition in your tests, please convert these to a PhpUnit dataProvider. This enables us to determine if all usecases are covered